pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -569,12 +569,19 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyView model.ProxyView, se
 			},
 			Metadata: &core.Metadata{},
 		}
+		var metadata *model.EndpointMetadata
+		var istioCanonicalService bool
+		svcLabels := service.Attributes.Labels
+		if _, ok := svcLabels[model.IstioCanonicalServiceLabelName]; ok {
+			metadata = instance.Endpoint.MetadataClone()
```

Comment:
why do we clone here only? We may mutate on L583, and if L582 is not met this is wasted

Reply:
good point. changed